src/vivarium_testing_utils/automated_validation/data_transformation/age_groups.py

```diff
@@ -0,0 +1,517 @@
+from __future__ import annotations
+
+import re
+
+import pandas as pd
+from loguru import logger
+
+AGE_GROUP_COLUMN = "age_group"
+AGE_START_COLUMN = "age_start"
+AGE_END_COLUMN = "age_end"
+
+AgeTuple = tuple[str, int | float, int | float]
+AgeRange = tuple[int | float, int | float]
+
+
+class AgeGroup:
+    """
+    Class to represent a single age interval with start and end ages.
+    """
+
+    def __init__(self, name: str, start: int | float, end: int | float):
+        """
+
+        Parameters
+        ----------
+        name
+            The name of the age group.
+        start
+            The start age of the age group.
+        end
+            The end age of the age group.
+        Raises
+        ------
+        ValueError
+            If the start or end age is negative, or if the end age is less than or equal to the start age.
+        """
+        self.name = name
+        if start < 0:
+            raise ValueError(f"Negative start age.")
```

Comment:
We actually did create a negative age start to account for birth prevalence in MNCNH. I think that was a hack that we may have gotten rid of and may not want to revisit. Just flagging here though.

Reply:
We got rid of that so there is exposure and birth exposure artifact keys replicating birth prevalence. I think it is important to note that those keys for the birth age group do not have age start or end in the index.